### PR TITLE
Fix dropdown assertions

### PR DESCRIPTION
**Pull request by AaronVanGeffen**
Opening a company selection dropdown, sets the highlighted item to -1. The `size_t` type is unsigned, so that would be interpreted as bigger than the uint8_t max.

### CODE REVIEW

File: src/OpenLoco/Ui/Dropdown.cpp [hunk 1]
```diff
@@ -122,21 +122,24 @@ namespace OpenLoco::Ui::Dropdown
 
     void setItemDisabled(size_t index)
     {
-        assert(index < std::numeric_limits<uint8_t>::max());
+        // Ensure that a valid item index is passed, or -1 to disable.
+        assert(index < std::numeric_limits<uint8_t>::max() || index == std::numeric_limits<size_t>::max());
```
Comment (duncanspumpkin) [inline src/OpenLoco/Ui/Dropdown.cpp:126]: This change doesn't make sense. You can't disable with -1.

File: src/OpenLoco/Ui/Dropdown.cpp [hunk 2]
```diff
@@ -122,21 +122,24 @@ namespace OpenLoco::Ui::Dropdown
...
 
         _dropdownHighlightedIndex = static_cast<uint8_t>(index);
     }
 
     void setItemSelected(size_t index)
     {
-        assert(index < std::numeric_limits<uint8_t>::max());
+        // Ensure that a valid item index is passed, or -1 to disable.
+        assert(index < std::numeric_limits<uint8_t>::max() || index == std::numeric_limits<size_t>::max());
```
Comment (duncanspumpkin) [inline src/OpenLoco/Ui/Dropdown.cpp:142]: Again this is invalid

**Review by duncanspumpkin — CHANGES_REQUESTED** [review]
Only the setHighlightedItem changes make sense. You're going to cause all sorts of issues left shifting by >32 let alone 0xFF

**Comment by AaronVanGeffen** [comment]
You're right. I'll revert those changes.

**Review by duncanspumpkin — APPROVED** [review]

**Pull request merged**